src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -241,10 +243,19 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
       return false;
     }
     if (pattern != null) {
-      String path  = request.path();
+      String path = request.path();
       if (mountPoint != null) {
         path = path.substring(mountPoint.length());
       }
+
+      // decode the path as it could contain escaped chars.
+      try {
+        path = URLDecoder.decode(path, "UTF-8");
```

Comment:
question: is it always UTF-8 ? It may be the server encoding no ? (actually, I've no clue).